modules/base/rendering/renderablemodel.cpp

```diff
@@ -297,17 +297,14 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
 
     addProperty(Fadeable::_opacity);
 
-    if (p.forceRenderInvisible.has_value()) {
-        _forceRenderInvisible = *p.forceRenderInvisible;
-
-        if (!_forceRenderInvisible) {
-            // Asset file have specifically said to not render invisible parts,
-            // do not notify in the log if invisible parts are detected and dropped
-            _notifyInvisibleDropped = false;
-        }
+    _forceRenderInvisible = p.forceRenderInvisible.value_or(_forceRenderInvisible);
+    if (p.forceRenderInvisible.has_value() && !_forceRenderInvisible) {
+        // Asset file have specifically said to not render invisible parts, do not notify
+        // in the log if invisible parts are detected and dropped
+        _notifyInvisibleDropped = false;
     }
 
-    _file = absPath(p.geometryFile);
+    _file = p.geometryFile;
```

Comment:
I guess this could be considered a breaking change.  Now the user would have to do the `openspace.absPath` call. Doesn't look like it's an issue with any of our models though

Just want to ask - is this how we want all our file inputs in assets to work?  Thinking that there may be a mix as of now

Reply:
It's not a breaking change since the parameters struct already accepted a `std::filesystem::path` before;  the FileVerifier (verifier.cpp:256) will reject an input that is not a valid, existing file. So a path like `${DATA}/test.jpg` would have been rejected even before we got to this line, already making the `openspace.absPath` call necessary in the asset file.   So the `absPath` call in here will always return the same functional path (it will remove some indirections, like    `C:/abc/def/../ghi` ->`C:/abc/ghi`  but I wouldn't consider this a breaking change if we nolonger remove those

And yes, we want our asset files to provide a valid path to the dictionary.  Almost all of the time we are not using path tokens in the file path we are paying the cost of an abspath every time

Reply:
Gotcha